packages/embark-deployment/src/index.js

```diff
@@ -69,6 +69,20 @@ class DeployManager {
                 if (typeof result === 'function') {
                   callback = result;
                 }
+                if (contract.address && typeof contract.address === 'function') {
+                  contract.addressHandler = contract.address;
+                  delete contract.address;
```

Comment:
That's not exactly the place I was thinking that you should put the check.
I was thinking of checking that in the contract-manager so that as soon as we load the contract config, we remove the function from `address` in case it clashes somewhere and to avoid to do checks like you did below where you had to check if  address is a string (it should always be a string pretty much)
Somewhere like here: https://github.com/embark-framework/embark/blob/master/packages/embark-contracts-manager/src/index.js#L380